### PR TITLE
KLB segmentation

### PR DESCRIPTION
**Pull request by jburel**
* add notebook showing how to analyse in parallel using dask-image
* Add corresponding python script. When running the python script, **change** the number of ``z ``and ``t``. it is currently set to ``sizeZ`` and ``sizeT``.

 * Add a doc page to show how to analyze in parallel

### CODE REVIEW

File: docs/analyze_dask_s3.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+----------------
+
+We will show how to use `dask <https://dask.org/>`_ to analyze an IDR image
+stored in a public S3 repository
+
+We will show:
+
+- How to connect to IDR to retrieve the image metadata
+
+- How load the Zarr binary stored in a public repository
```
Comment (mtbc) [inline docs/analyze_dask_s3.rst]:
```suggestion
- How to load the Zarr binary stored in a public repository
```

File: notebooks/zarr-public-s3-segmentation-parallel.ipynb [hunk 2]
```diff
@@ -0,0 +1,423 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Example of analyzing in paraller a Zarr image from a public S3 repository\n",
```
Comment (mtbc) [inline notebooks/zarr-public-s3-segmentation-parallel.ipynb]:
```suggestion
    "# Example of analyzing in parallel a Zarr image from a public S3 repository\n",
```

**Comment by pwalczysko** [comment]
Failed travis

**Comment by pwalczysko** [comment]
repo2docker

The notebook works well and fast. I just have no idea what it really does. Some segmentation ? What are the 10 result images I am getting ? Some segmentation ? Which idr study number am I loading ? (id of the image is not enough to identify)

**Comment by pwalczysko** [comment]
> change the number of z and t. it is currently set to sizeZ and sizeT.

Change the number to what ? I will try somehing, but atm, no idea what and how that will pan out.

**Comment by jburel** [comment]
change to a low number e.g. 2, 2, 1 so you do not wait forever

**Comment by pwalczysko** [comment]
```
(notebook) pwalczysko@427963b5249e:~/scripts$ vi public_s3_segmentation_parallel.py
bash: vi: command not found
(notebook) pwalczysko@427963b5249e:~/scripts$
```
How to edit ?

**Comment by jburel** [comment]
Link to the study is in the header of the notebook.
Sub-section indicates that a segmentation is performed

**Comment by jburel** [comment]
you can edit via the UI by opening the script, and click save
Vi is not installed

**Comment by pwalczysko** [comment]
> you can edit via the UI by opening the script, and click save
Vi is not installed

Thanks that worked.

The script worked too. Except, again, no idea what does its output mean.

<details>

```
dask.array<from-zarr, shape=(532, 2, 988, 128, 135), dtype=>u2, chunksize=(1, 1, 1, 128, 135), chunktype=numpy.ndarray>
t:2, c: 0, z:1
t:2, c: 0, z:2
t:0, c: 0, z:4
t:4, c: 0, z:0
t:4, c: 0, z:3
t:3, c: 0, z:2
t:1, c: 0, z:2
t:0, c: 0, z:0
t:1, c: 0, z:3
t:0, c: 0, z:3
t:2, c: 0, z:0
t:3, c: 0, z:3
t:3, c: 0, z:0
t:4, c: 0, z:1
t:2, c: 0, z:4
t:1, c: 0, z:1
t:0, c: 0, z:2
t:0, c: 0, z:1
t:4, c: 0, z:2
t:2, c: 0, z:3
t:1, c: 0, z:0
t:3, c: 0, z:1
t:4, c: 0, z:4
t:3, c: 0, z:4
t:1, c: 0, z:4
4.244684219360352
done
```

</details>

**Comment by jburel** [comment]
@pwalczysko I have improved the print statements
```
Connected: True
Dask array: dask.array<from-zarr, shape=(532, 2, 988, 128, 135), dtype=>u2, chunksize=(1, 1, 1, 128,135), chunktype=numpy.ndarray>
Plane coordinates: t:0, c: 0, z:0
Compute time: 0.4310469627380371
done
```

**Comment by pwalczysko** [comment]
> Compute time: 0.4310469627380371

What are the units here ?

Also, I understood that something is run in parallel here. How many threads are there running in parallel ? The script (unlike the notebook) has also the disadvantage that no images are being delivered by it - where are my segmentations (or what did the script actually achieve ?)

**Comment by jburel** [comment]
@pwalczysko We could save the results on disk. What do you think?

**Comment by pwalczysko** [comment]
> @pwalczysko We could save the results on disk. What do you think?

Yes, possibly. How big are the results ? I understand that the dataset is the KLB from the idr , but do not know the accession nr. by heart

**Comment by jburel** [comment]
We can save the first 5.

**Comment by jburel** [comment]
@pwalczysko the last commit now saves the first 5 results as png

**Comment by pwalczysko** [comment]
> We can save the first 5.

Yes, sorry, that worked as expected. But, the images are very underwhelming, possibly because they are from the "edge" images of the stack ?
image_t_0_c_0_z_2

**Comment by pwalczysko** [comment]
I tried this

```
# Prepare-call
def prepare_call(image):
    number_t = 106  # reduce for demo
    number_z = 106
    number_c = 1
    lazy_results = []
    for t in range(100,number_t):
        for z in range(100,number_z):
            for c in range(number_c):
```
to get deeper into the stack. 
But the images look exactly the same. They seem to have uniform intensity values all over.

**Comment by jburel** [comment]
The notebook starts from the middle z and t and add the z/t value, otherwise you can see much, 106 is still too low considering the number of z-sections/timepoints (988 x 532)
So maybe we follow the same strategy for the script. mid-section and add z/t to it

**Comment by pwalczysko** [comment]
> So maybe we follow the same strategy for the script. mid-section and add z/t to it

I think yes

**Comment by jburel** [comment]
Both notebook and script have range around mid-plane, like this you can see something

**Comment by pwalczysko** [comment]
Great, now I can see something on the 5 saved images in the script.

Ready to merge fmpov

**Pull request merged**